src/main/java/xyz/nucleoid/plasmid/game/GameWorld.java

```diff
@@ -94,33 +130,65 @@ public void setGame(Game game) {
         this.invoker(GameOpenListener.EVENT).onOpen();
     }
 
+    /**
+     * Closes this {@link GameWorld}.
+     *
+     * <p>All {@link GameCloseListener#EVENT} listeners are notified of the close, and then all listeners & rules for this {@link GameWorld} are reset.
+     */
     private void closeGame() {
         this.invoker(GameCloseListener.EVENT).onClose();
-        this.listeners = new EventListeners();
-        this.rules = GameRuleSet.empty();
+        this.game = new Game();
     }
 
+    /**
+     * Attempts to add the given {@link ServerPlayerEntity} to this {@link GameWorld}.
```

Comment:
I think also worth not mentioning bubble worlds here and rather describing the function: a player *must* be added by this function, or they cannot exist in the world. Also can describe that it does not generally need to be manually called, because it's already controlled by the player offer abstraction